code/modules/power/supermatter/supermatter.dm

```diff
@@ -491,6 +491,10 @@
 		investigate_log("has consumed [key_name(user)].", "supermatter")
 		user.dust()
 		matter_power += 200
+	else if(istype(AM, /obj/singularity))
+		var/obj/singularity/S = AM
+		S.consume(src)
```

Comment:
I would just return, so that the singularity doesn't accidentally double-consume the supermatter.